manifests/webhook/config.pp

```diff
@@ -6,51 +6,52 @@
-  $use_mcollective                = $r10k::params::webhook_use_mcollective,
-  $r10k_deploy_arguments          = $r10k::params::webhook_r10k_deploy_arguments,
-  $public_key_path                = $r10k::params::webhook_public_key_path,
-  $private_key_path               = $r10k::params::webhook_private_key_path,
-  $yaml_template                  = $r10k::params::webhook_yaml_template,
-  $command_prefix                 = $r10k::params::webhook_r10k_command_prefix,
-  $repository_events              = $r10k::params::webhook_repository_events,
-  $allow_uppercase                = $r10k::params::webhook_allow_uppercase,
-  $slack_webhook                  = $r10k::params::webhook_slack_webhook,
-  $slack_channel                  = $r10k::params::webhook_slack_channel,
-  $slack_username                 = $r10k::params::webhook_slack_username,
-  $slack_proxy_url                = $r10k::params::webhook_slack_proxy_url,
-  $rocketchat_webhook             = $r10k::params::webhook_rocketchat_webhook,
-  $rocketchat_channel             = $r10k::params::webhook_rocketchat_channel,
-  $rocketchat_username            = $r10k::params::webhook_rocketchat_username,
-  $configfile_owner               = $r10k::params::webhook_configfile_owner,
-  $configfile_group               = $r10k::params::webhook_configfile_group,
-  $configfile_mode                = $r10k::params::webhook_configfile_mode,
-  $configfile                     = '/etc/webhook.yaml',
-  $manage_symlink                 = false,
-  $configfile_symlink             = '/etc/webhook.yaml',
-  $enable_mutex_lock              = $r10k::params::webhook_enable_mutex_lock,
-  Array $ignore_environments      = $r10k::params::webhook_ignore_environments,
-  Optional[String] $mco_arguments = $r10k::params::webhook_mco_arguments,
-  Boolean $generate_types         = $r10k::params::webhook_generate_types,
+  $ensure                            = true,
+  Variant[String, Hash] $hash        = 'UNSET',
+  $certname                          = $r10k::params::webhook_certname,
+  $certpath                          = $r10k::params::webhook_certpath,
+  $user                              = $r10k::params::webhook_user,
+  $pass                              = $r10k::params::webhook_pass,
+  $bind_address                      = $r10k::params::webhook_bind_address,
+  $port                              = $r10k::params::webhook_port,
+  $access_logfile                    = $r10k::params::webhook_access_logfile,
+  $client_cfg                        = $r10k::params::webhook_client_cfg,
+  $default_branch                    = $r10k::params::webhook_default_branch,
+  $use_mco_ruby                      = $r10k::params::webhook_use_mco_ruby,
+  $protected                         = $r10k::params::webhook_protected,
+  $github_secret                     = $r10k::params::webhook_github_secret,
+  Optional[String] $bitbucket_secret = $r10k::params::webhook_bitbucket_secret,
```

Comment:
Can you enforce that empty strings are not allowed?
```suggestion
  Optional[String[1]] $bitbucket_secret = $r10k::params::webhook_bitbucket_secret,
```

I think that would make sense.